### PR TITLE
Adding aqt config needed for inference to all layers.

### PR DESCRIPTION
**Pull request by singh-mitali**
- Adds more flexible option to pass config quantization and not just int8
- Adds support for passing quantization mode which is needed to freeze/unfreeze params

### CODE REVIEW

File: MaxText/layers/attentions.py [hunk 1]
```diff
@@ -82,25 +83,22 @@ def apply_mask_to_logits(logits: Array, mask: Array):
   """
   return jnp.where((mask >= DEFAULT_MASK_VALUE * 0.5), logits, DEFAULT_MASK_VALUE)
 
-def _maybe_aqt_einsum(int8_training, aqt_rng):
+def _maybe_aqt_einsum(quant: Quant):
```
Comment (rwitten) [inline MaxText/layers/attentions.py:86]: Can we validate that the quantization strategy matches our expectations here? Just what we were talking about hahaha

**Review by rwitten — APPROVED** [review]
LGTM but please get Matt's approval as well.

File: MaxText/layers/attentions.py (hunk 1 above)
Comment (singh-mitali) [reply MaxText/layers/attentions.py:86]: This change makes the code flexible to add  new strategies. Currently only added int8  but can easily add any others to quantizarion.py.

File: MaxText/layers/quantization.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  raise ValueError(f'Invalid value configured for quantization {quant_str}.')
+
+
+def configure_quantization(config: Config, quant_mode_str: str = 'train'):
+  """ Configure quantization based on user config and quant mode."""
+  quant_cfg = _get_quant_config(config.quantization)
+  if quant_cfg:
+    if quant_mode_str == 'train':
+      return AqtQuantization(quant_cfg, aqt_flax.QuantMode.DYNAMIC)
+    elif quant_mode_str  =='serve':
```
Comment (gobbleturk) [inline MaxText/layers/quantization.py:56]: nit: whitespace around ==

File: MaxText/layers/quantization.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      return AqtQuantization(quant_cfg, aqt_flax.QuantMode.FREEZE)
+    elif quant_mode_str  =='convert':
```
Comment (gobbleturk) [inline MaxText/layers/quantization.py:58]: nit: same whitespace

File: MaxText/tests/quantization_test.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+""" Tests for the quantizations """
+from jax import numpy as jnp
+from jax import random, lax
+from flax import linen as nn
+import numpy as np
+import pyconfig
+from layers import quantization
+import unittest
+
+class ABC(nn.Module):
```
Comment (gobbleturk) [inline MaxText/tests/quantization_test.py:26]: nit: Why is the name of this class ABC? Should this be AqtLayerTest or similar?

**Review by gobbleturk — APPROVED** [review]

File: MaxText/layers/quantization.py (hunk 2 above)
Comment (singh-mitali) [reply MaxText/layers/quantization.py:56]: Done

File: MaxText/layers/quantization.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+"""Quantization library."""
+
+import functools
+
+from aqt.jax.v2 import config as aqt_config
+from aqt.jax.v2.flax import aqt_flax
+from common_types import Config
+from dataclasses import dataclass
+
+AQT_INT8_CONFIG = aqt_config.fully_quantized()
```
Comment (gobbleturk) [inline MaxText/layers/quantization.py:24]: We should keep this as config_v3

**Review by gobbleturk — CHANGES_REQUESTED** [review]

File: MaxText/layers/quantization.py (hunk 5 above)
Comment (singh-mitali) [reply MaxText/layers/quantization.py]: Done

**Review by gobbleturk — APPROVED** [review]

**Comment by mitalisi** [comment]
Created a new pull request with updated aqt code.

**Pull request closed without merging**